### PR TITLE
fix(aws-lambda-nodejs): Remove dependency on global parcel installation

### LINKED ISSUE #6204: aws-lambda-nodejs doesn't build if parcel is not installed globally
I think because of [this line](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-lambda-nodejs/lib/build.ts#L75), parcel needs to be installed globally for aws-lambda-nodejs to build.

### Reproduction Steps

1. Take a working app containing a aws-lambda-nodejs construct.
2. Run `npm remove -g parcel`
3. Run `cdk synth` on the app. The command fails with:

    > Failed to build file at ./lambda/src/index.ts: Error: spawnSync parcel ENOENT

### Error Log

See above

### Environment

  - **CLI Version      : 1.22.0 (build 309ac1b)
  - **Framework Version: 1.23.0
  - **OS               : MacOS 10.15.3
  - **Language         : typescript

### Other

I think you need to reference the binary as a dependency, similar to how npx works, rather than invoking spawn

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by mattsains**
Fixes #6204

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by mattsains** [comment]
I'm no longer sure whether this would fix the problem, the spawn command might mean that npx is running on the package using aws-lambda-nodejs rather than aws-lambda-nodejs itself. I'm trying to consume my local version of the CDK to see if it works.

File: packages/@aws-cdk/aws-lambda-nodejs/lib/build.ts [hunk 1]
```diff
@@ -72,7 +73,7 @@ export function build(options: BuildOptions): void {
         : [],
     ].filter(Boolean) as string[];
 
-    const parcel = spawnSync('parcel', args);
+    const parcel = spawnSync('npx', args);
```
Comment (jogold) [inline packages/@aws-cdk/aws-lambda-nodejs/lib/build.ts]:
```suggestion
    const parcelPkgPath = require.resolve('parcel-bundler/package.json'); // eslint-disable-line @typescript-eslint/no-require-imports
    const parcelDir = path.dirname(parcelPkgPath);
    const parcelPkg = require(parcelPkgPath); // eslint-disable-line @typescript-eslint/no-require-imports
    const binPath = path.join(parcelDir, parcelPkg.bin.parcel);
    
    const parcel = spawnSync(binPath, args);
```

File: packages/@aws-cdk/aws-lambda-nodejs/test/function.test.ts [hunk 2]
```diff
@@ -31,7 +31,8 @@ test('NodejsFunction', () => {
   // THEN
   const { spawnSync } = require('child_process'); // eslint-disable-line @typescript-eslint/no-require-imports
 
-  expect(spawnSync).toHaveBeenCalledWith('parcel', expect.arrayContaining([
+  expect(spawnSync).toHaveBeenCalledWith('npx', expect.arrayContaining([
```
Comment (jogold) [inline packages/@aws-cdk/aws-lambda-nodejs/test/function.test.ts]:
```suggestion
  expect(spawnSync).toHaveBeenCalledWith(expect.stringContaining('parcel-bundler'), expect.arrayContaining([
```

File: packages/@aws-cdk/aws-lambda-nodejs/test/function.test.ts [hunk 3]
```diff
@@ -50,7 +51,8 @@ test('NodejsFunction', () => {
   ]));
 
   // Automatically finds .js handler file
-  expect(spawnSync).toHaveBeenCalledWith('parcel', expect.arrayContaining([
+  expect(spawnSync).toHaveBeenCalledWith('npx', expect.arrayContaining([
```
Comment (jogold) [inline packages/@aws-cdk/aws-lambda-nodejs/test/function.test.ts]:
```suggestion
  expect(spawnSync).toHaveBeenCalledWith(expect.stringContaining('parcel-bundler'), expect.arrayContaining([
```

File: packages/@aws-cdk/aws-lambda-nodejs/lib/build.ts (hunk 1 above)
Comment (jogold) [reply packages/@aws-cdk/aws-lambda-nodejs/lib/build.ts]: @eladb can you have a look?

Comment (mattsains) [reply packages/@aws-cdk/aws-lambda-nodejs/lib/build.ts]: I will try this out, thanks for the suggestion!

Comment (mattsains) [reply packages/@aws-cdk/aws-lambda-nodejs/lib/build.ts]: that worked for me, and I think it's the right way to go, I'll update this PR

Comment (eladb) [reply packages/@aws-cdk/aws-lambda-nodejs/lib/build.ts]: Yes this is how I would do it

Comment (mattsains) [reply packages/@aws-cdk/aws-lambda-nodejs/lib/build.ts]: I updated the PR with that suggestion.

**Pull request merged**